scripts/pre_commit_linter.py

```diff
@@ -342,10 +361,57 @@ def _pre_commit_linter():
     summary_messages.append(py_result.get())
     print '\n'.join(summary_messages)
     print ''
+    return summary_messages
+
+
+# This part is used for detecting bad patterns.
+def _check_bad_patterns(all_files, summary_messages):
+    print 'Starting Pattern Checks'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    for filename in all_files:
+        with open(filename) as f:
+            content = f.read()
+            total_files_checked += 1
+            failed = False
+            for pattern in BAD_PATTERNS:
+                if pattern in content:
+                    failed = True
+                    print '%s --> %s' % (
+                        filename, BAD_PATTERNS[pattern])
+                    total_error_count += 1
+            if failed:
+                summary_messages.append('%s   Pattern checks failed'
+                                        % _MESSAGE_TYPE_FAILED)
+            else:
+                summary_messages.append('%s   Pattern checks passed'
+                                        % _MESSAGE_TYPE_SUCCESS)
+
+    print ''
+    print '----------------------------------------'
+    print ''
+    if total_files_checked == 0:
+        print "There are no files to be checked."
+    else:
+        print '(%s files checked, %s errors found)' % (
+            total_files_checked, total_error_count)
+        print summary_messages[len(summary_messages)-1]
```

Comment:
This is ugly, since it requires too much contextual knowledge about the structure of summary_messages. I think the easiest solution is to just put the 'print' statement alongside when you append the message to summary_messages.